tests/api-compatibility/acceptable-breakages-v10.0.txt

```diff
@@ -0,0 +1,40 @@
+Compat issues with assembly Mono.Android:
+MembersMustExist : Member 'Android.Media.AudioRecord.add_RoutingChanged(System.EventHandler<Android.Media.AudioRecord.RoutingChangedEventArgs>)' does not exist in the implementation but it does exist in the contract.
```

Comment:
...and I find this both horrifying and confusing: this sounds like a breaking change, yet it's not listed in https://github.com/xamarin/xamarin-android-api-compatibility/blob/master/inter-api-extra-v9.0-v10.0.txt nor in https://github.com/xamarin/xamarin-android-api-compatibility/commit/336eedaa51883d90100e30c95a007d8922ab19f0 (which must be viewed locally, as it's too big for GitHub to display), and I don't remember it coming up in the context of breaking changes previously.

```
$ git show 336eedaa51883d90100e30c95a007d8922ab19f0 | grep -i RoutingChanged
# no output
```

Yet, when I diff, it's there:

```diff
$ diff -u src/Mono.Android/obj/Debug/android-28/mcw/Android.Media.AudioRecord.cs src/Mono.Android/obj/Debug/android-29/mcw/Android.Media.AudioRecord.cs
-               public event EventHandler<Android.Media.AudioRecord.RoutingChangedEventArgs> RoutingChanged {
+               public event EventHandler<Android.Media.AudioRoutingOnRoutingChangedEventArgs> RoutingChanged {
```

Why didn't our previous tooling find this?

For that matter, this finds nothing within the xamarin-android-api-compat repo:

```
$ grep '<event.*RoutingChanged' reference/Mono.Android.xml
# no matches
```

(Feeling of horror increases...)

Reply:
It took me quite a bit to remember this, but that's expected!
We were wrongly not generating that event, so that's why previous tool never complained about it.

see this: https://github.com/xamarin/xamarin-android/commit/bcae9415f188ae0f602bab89c57db1c265fb36d0#diff-14048fd4cb6595cd0ad8cf1b41175ef5L1301

After my fix, we started generating that but only until api 28

Reply:
> We were wrongly not generating that event

OK, so it didn't previously exist

> After my fix, we started generating that but only until api 28

...and this is where we made a mistake.  If we weren't previously emitting it, we should have *continued* not emitting it, so that when we "introduced" it in API-29, it wouldn't be incompatible with previous versions.

😞

Reply:
We discussed the issue together. 
The bigger issue was that the handlers/delegate were generated, but not the event. So the api was "half" there.
There was a bug: https://github.com/xamarin/xamarin-android/issues/3313
were someone was trying to use it on API 24, if we did not fix it, then that person could not have his project working for API 24